Project Beats: add skip controls

sanchitmalhotra126 opened the pull request:
Adds outside of playback skip forward/backwards controls. These advance the playhead forwards or backwards by one measure. Also clicking a measure number or the measures header background will move the playhead to that position outside of playback. To show the current start position, we show the playhead outside of playback in a neutral light color, which switches to the current active yellow when playback starts.

https://github.com/code-dot-org/code-dot-org/assets/85528507/1ab4b981-5df2-4525-98dc-ed55ccc194e8

## Links

[Music Lab Functional Spec](https://docs.google.com/document/d/1vssq4yGvuP2XbI71-drzqNdmNC37pZh46oyqRFee618/edit)

## Testing story

Tested locally.

levadadenys reviewed (approved):
LGTM!
Wow, the progress with Music Lab is awesome. Looks like web version of Ableton/FL studio to me (which is awesome 😄) already or at least is getting closer and closer to it

breville commented:
Very cool.  Some initial UI thoughts:

- I really liked the `Run` button being the full width of the panel.  Could we possible put these buttons in the empty space below the triggers?
- With white backgrounds, these buttons feel more prominent than then `Run` button.  Could we tone them down or consider purple?
- We should disable them when playing, since they don't do anything.
- Would it be worth having a third button which jumps the playhead back to the beginning?  
- This exposes the deeper issue that we don't support playing a sample that's not at its beginning.  Most noticeably, if you start from one bar into the song, but the song comprises two-bar long samples (which we use a lot), then you don't hear that sound at all.  Should we consider updating our audio implementation to support starting a sound from partway through?

breville commented:
One other thought: should we parameterize the appearance of this feature?  It feels like we might want to introduce it at a later point in a tutorial sequence, when things start to get more complex.

breville commented:
One potential issue: after hitting stop, the highlighted Blockly blocks don't unhighlight.

sanchitmalhotra126 commented:
@breville revisiting this PR, addressing the feedback we discussed

> * after hitting stop, the highlighted Blockly blocks don't unhighlight.

this should be fixed now!

> * for the disabled buttons, we can try making the existing ones `opacity: 50%`.

Done

>One other thought: should we parameterize the appearance of this feature? It feels like we might want to introduce it at a later point in a tutorial sequence, when things start to get more complex.

This feature is now hidden behind a config flag (`skip-controls=(enabled|disabled)`). I've also gone ahead and put keyboard shortcuts behind a config flag as well (`keyboard-shortcuts=(enabled|disabled)`). Both will default to disabled if not specified.

breville commented on apps/src/musicMenu/MusicMenu.jsx:
```diff
@@ -82,6 +82,22 @@ const optionsList = [
+    name: 'skip-controls',
+    type: 'radio',
+    values: [
+      {value: 'disabled', description: 'Disable skip controls.'},
+      {value: 'enabled', description: 'Enable skip controls.'},
```
super nit: we use `true` and `false` for other values in the menu.  `enabled` and `disabled` are definitely nice, but should we be consistent with whichever we choose?  One factor might be what works best in the context of levelbuilder parameters.

sanchitmalhotra126 replied:
Oh good call, I'll make this `true`/`false` for consistency.

breville reviewed (approved):
Great feature!
